### ISSUE TITLE
Question: StringGrid

### ISSUE THREAD

**Issue by halfnibble** [pywinauto/pywinauto#271]
I have a question, and I'm not sure where to look.

Let's say I need to get data from a StringGrid (win32 backend). It's a lot like a ListBox.

Can I simply create a new control type in pywinauto that mimics ListBox, or would this require hacking the win32 backend?

Thanks in advance.

**Comment by vasily-v-ryabov** [comment]
No special hacking required. Just create a wrapper similar to ListBoxWrapper.
Example:
```python
class StringGridWrapper(hwndwrapper.HwndWrapper):

    """Wrap a StringGrid control"""

    friendlyclassname = "StringGrid"
    windowclasses = [
        "TStringGrid",
        ".*StringGrid", ]
    has_title = False

    # methods...
```
Every control matching one of `windowclasses` will be automatically registered as a StringGrid control. The only thing you need to keep in mind that the wrapper class must be placed in one of the files: `win32_controls.py`, `common_controls.py` or `hwndwrapper.py`. Another way is to create some `custom_controls.py` and to import it in `.\pywinauto\controls\__init__.py` (this will register the wrapper in the backend). Registering new wrappers explicitly (without pywinauto's modification) isn't implemented yet. But we appreciate pull requests. ;)

**Comment by majj** [comment]
class MSFlexGrid has the same issue.

**Comment by halfnibble** [comment]
Thanks @vasily-v-ryabov for the excellent response.

This is exactly what I needed. You may assign this issue to me, and I'll create a pull request to add your explanation to the documentation.

**Comment by vasily-v-ryabov** [comment]
Great! Example of custom control implementation would be really interesting. I had some custom controls at my previous job, but could not publish them. Thanks for making the docs better. 👍

**Comment by halfnibble** [comment]
@vasily-v-ryabov I have another question. A product of being a *nix guy.

Let's say I want to define a new method that sends a different message. For example, RowCount(). I'm looking in win32defines.py, and I'm wondering if these constants are documented somewhere? Thanks.

**Comment by vasily-v-ryabov** [comment]
Window message ids are often well documented in MSDN.
Example: [TCM_GETROWCOUNT](https://msdn.microsoft.com/en-us/library/windows/desktop/bb760596(v=vs.85).aspx).

BTW, I'd prefer having new methods in PEP8 style only (`RowCount = row_count` is not necessary). It was done for partial backward compatibility with 0.5.4.

**Comment by airelil** [comment]
@halfnibble, there are many constants that are already defined in win32con (pywin32 package). See #265. So you'd better try to find a definition in win32con.

**Comment by vasily-v-ryabov** [comment]
Sorry for not saying this before. I remember I've tried to support `TStringGrid` control from Delphi but could not find appropriate window messages. They are quite non-standard probably. You may try catching window messages by Spy++ though I have no luck here as well. `backend="uia"` seems useless here too. It would be really cool, if you were succeeded.

**Comment by halfnibble** [comment]
@vasily-v-ryabov You are correct. TStringGrid has no appropriate win32 API messages. It may be possible with OCR, but this project has a budget. I will add a note to the documentation PR that discusses the potential for unsupported UI components.

**Comment by vasily-v-ryabov** [comment]
Not sure it's applicable for your project, but if have an access to the source code, [the following control wrappers](https://github.com/jhc-systems/DelphiUIAutomation/tree/master/controls) add MS UI Automation to many Delphi controls including StringGrid. In case of success you need to use `backend='uia'`.

**Comment by wxl200854** [comment]
hope support stringgrid
